test/chart-template.js

```diff
@@ -0,0 +1,392 @@
+        } finally {
+            this.ready = true;
+        }
+
+    };
+
+    /* *
+     *
+     *  Static Functions
+     *
+     * */
+
+    /**
+     * Prepares a chart template for a test. This function works asynchronously.
+     *
+     * @param {string} name
+     * The reference name of the template to prepare for the test
+     *
+     * @param {object|undefind} chartOptions
+     * The additional options to customize the chart of the template
+     *
+     * @param {function} testCallback
+     * The callback with the prepared chart template as the first argument
+     *
+     * @return {void}
+     */
+    ChartTemplate.test = function (name, chartOptions, testCallback) {
+
+        if (typeof name !== 'string' ||
+            typeof chartOptions !== 'object' ||
+            typeof testCallback !== 'function'
+        ) {
+            throw new Error('Arguments are invalid');
+        }
+
+        var chartTemplate = chartTemplates[name];
+
+        if (!chartTemplate) {
+            throw new Error('Template "' + name + '" is not registered');
+        }
+
+        chartOptions = (chartOptions || {});
+
+        if (!(chartTemplate instanceof ChartTemplate)) {
+            chartTemplates[name] = new ChartTemplate(
+                chartTemplate.name,
+                chartTemplate.chartConstructor,
+                chartTemplate.chartOptions
+            );
+            chartTemplate = chartTemplates[name];
+        }
+
+        chartTemplate.test(chartOptions, testCallback);
+
+    };
+
+    /**
+     * Registers a chart template for additional tests
+     *
+     * @param {string} name
+     * The reference name of the chart
+     *
+     * @param {Highcharts.Chart} chartConstructor
+     * The chart constructor function for the template
+     *
+     * @param {object} chartOptions
+     * The default chart Options for the template
+     *
+     * @return {void}
+     */
+    ChartTemplate.register = function (name, chartConstructor, chartOptions) {
+
+        if (typeof name !== 'string' ||
+            typeof chartConstructor !== 'function' ||
+            typeof chartOptions !== 'object'
+        ) {
+            throw new Error('Arguments are invalid');
+        }
+
+        if (chartTemplates[name]) {
+            throw new Error('Chart template already registered');
+        }
+
+        chartTemplates[name] = {
+            name,
+            chartConstructor,
+            chartOptions
+        };
+
+    };
+
+    /* *
+     *
+     *  Static Properties
+     *
+     * */
+
+    /**
+     * The registered chart templates
+     *
+     * @type {Array<ChartTemplate>}
+     */
+    Object.defineProperty(ChartTemplate, 'templates', {
+        configurable: false,
+        enumerable: true,
+        get: function () {
+            return chartTemplates;
+        }
+    });
+
+    /* *
+     *
+     *  Global
+     *
+     * */
+
+    // Prevent changes to ChartTemplate properties
+    Object.freeze(ChartTemplate);
```

Comment:
Similar to `defineProperty`: https://caniuse.com/#search=freeze (no support for IE < 9)